tests/main/views/test_frameworks.py

```diff
@@ -378,6 +395,158 @@ def test_update_framework_catches_db_errors(self, db_commit, open_example_framew
+        super(TestUpdateFrameworkPending, self).setup()
+
+        self.framework = Framework(slug='g-cloud-10', name='G-Cloud 10', framework='g-cloud', status='open',
+                                   clarification_questions_open=False, allow_declaration_reuse=True)
+        db.session.add(self.framework)
+        db.session.commit()
+
+        expected_info = self.EXPECTED_FRAMEWORK_DECLARATION_SUPPLIER_INFO
+        db.session.add(Supplier(supplier_id=1,
+                                name=expected_info["supplierTradingName"], description='',
+                                organisation_size=expected_info['supplierOrganisationSize'],
+                                trading_status=expected_info['supplierTradingStatus'],
+                                vat_number=expected_info['supplierVatNumber'],
+                                duns_number=expected_info['supplierDunsNumber'],
+                                registration_date=expected_info['supplierRegistrationDate'],
+                                other_company_registration_number=expected_info["supplierCompanyRegistrationNumber"],
+                                registration_country=expected_info["supplierRegisteredCountry"],
+                                registered_name=expected_info["supplierRegisteredName"],
+                                )
+                       )
+
+        db.session.add(ContactInformation(supplier_id=1,
+                                          contact_name='Contact for Supplier 1',
+                                          email='1@contact.com',
+                                          address1=expected_info['supplierRegisteredBuilding'],
+                                          city=expected_info['supplierRegisteredTown'],
+                                          postcode=expected_info['supplierRegisteredPostcode']
+                                          )
+                       )
+
+        db.session.add(SupplierFramework(framework_id=self.framework.id,
+                                         supplier_id=1,
+                                         declaration={}
+                                         )
+                       )
+
+        db.session.commit()
+
+    def teardown(self):
+        SupplierFramework.query.delete()
+        ContactInformation.query.delete()
+        Supplier.query.delete()
+
+        framework = Framework.query.filter(Framework.slug == "g-cloud-10").first()
+        if framework:
+            FrameworkLot.query.filter(FrameworkLot.framework_id == framework.id).delete()
+            Framework.query.filter(Framework.id == framework.id).delete()
+            db.session.commit()
+
+        super(TestUpdateFrameworkPending, self).teardown()
+
+    def _assert_declaration_matches(self, expected_declaration_data):
+        supplier_framework = SupplierFramework.query.filter(SupplierFramework.supplier_id == 1).first()
+        assert supplier_framework.declaration == expected_declaration_data
```

Comment:
I'm not very familiar with how the ContactInformation relates to the SupplierFramework model - do we need to query and assert the ContactInformation separately or does this cover it?

Reply:
This should cover it. SupplierFramework pulls in three fields from ContactInformation (address1, city, postcode), so when we assert a match on the declaration we're also asserting the appropriate contact information has been pulled in.